### PR TITLE
Refactoring of view handlers, to support parsing of single query instead of one per time bucket.

### PR DESCRIPTION
**Pull request by npawar**
Refactored view handlers to trigger just 1 query instead of multiple as per time buckets. The response is now parsed accordingly to extract data for each time bucket.
Added OTHER category for rows which fall below a certain configurable threshold.
Tested to check that tabular, contributors, heatmap and timeseries work correctly. 
The changes in dashboard response time for sample dataset 1 were as follows:
Tabular: from 4.71s to 277ms
Contributors: from 8.83s to 439ms, 9.59s to 1.2s
Heatmap: from 1.8s to 2.2s
Timeseries: from 2.39s to 228ms
For sample dataset 2:
Tabular: from 20s to 855ms
Heatmap: 6.20 to 13s
Contributors: 38s to 1.78s
Timeseries: 10s to 538ms

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeComparisonHandler.java [hunk 1]
```diff
@@ -145,64 +126,31 @@ public TimeOnTimeComparisonResponse handle(TimeOnTimeComparisonRequest compariso
...
+        Utils.computeMetricFunctionsFromExpressions(metricExpressions);
+    requestBuilder.setMetricFunctions(metricFunctionsFromExpressions);
+    return requestBuilder.build(requestReference);
   }
 
   public ThirdEyeClient getClient() {
     return queryCache.getClient();
   }
 
+  public static void main(String[] args) {
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeComparisonHandler.java]: remove this

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java [hunk 2]
```diff
@@ -1,250 +1,445 @@
...
-  public static Row parseAggregationOnlyResponse(TimeOnTimeComparisonRequest comparisonRequest,
-      Map<ThirdEyeRequest, ThirdEyeResponse> queryResultMap) {
-    Map<MetricFunction, Map<String, Double>> metricOnlyData =
-        TimeOnTimeResponseParser.processAggregateResponse(queryResultMap);
+        String dimensionName = baselineResponse.getGroupKeyColumns().get(1);
 
-    Row.Builder rowBuilder = new Row.Builder();
-    rowBuilder.setBaselineStart(comparisonRequest.getBaselineStart());
-    rowBuilder.setBaselineEnd(comparisonRequest.getBaselineEnd());
+        // other row
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java]: can you explain the logic, how many passes are we doing over the data to compute OTHER category

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java [hunk 3]
```diff
@@ -1,250 +1,445 @@
...
+          Row row = builder.build();
+          rows.add(row);
+        }
+
+        double[] baselineMetricSums = getMetricSums(baselineResponse);
+        double[] currentMetricSums = getMetricSums(currentResponse);
+        rows = getRowsWithThresholdFiltering(rows, dimensionName, metricFunctions, baselineMetricSums, currentMetricSums);
+
+      } else {
+        // no group by
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java]: throw unsupported exception, if it reaches here.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java [hunk 4]
```diff
@@ -1,250 +1,445 @@
...
-    List<MetricFunction> metricFunctions = thirdEyeResponse.getMetricFunctions();
-    for (MetricFunction metricFunction : metricFunctions) {
-      Map<String, String> row = thirdEyeResponse.getRow(metricFunction, 0);
-      // TODO THIS DEPENDS STRICTLY ON THE ORDER OF THE KEYS RETURNED?
-      String baselineValue = row.values().iterator().next();
-      if (!aggregateData.containsKey(metricFunction)) {
-        aggregateData.put(metricFunction, new HashMap<String, Double>());
-      }
-      aggregateData.get(metricFunction).put(baselineOrCurrent, Double.parseDouble(baselineValue));
+  List<Row> parseResponse() {
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/client/comparison/TimeOnTimeResponseParser.java]: extract each if/else block to a separate methods such as parseGroupByTimeResponse, parseGroupByTimeDimensionResponse parseGroupByDimensionResponse

**Comment by kishoreg** [comment]
LGTM. Please add the description and the perf improvement numbers

**Comment by npawar** [comment]
Approved by @kishoreg

**Pull request merged**